commands/gif.js

```diff
@@ -1,19 +1,47 @@
 const { giphy_api_key } = require("../config.js");
 const giphy = require("giphy-api")(giphy_api_key);
+/* 
+ * Giphy API Explorer: https://developers.giphy.com/explorer
+ * Giphy API Rest API docs: https://developers.giphy.com/docs/api/endpoint/
+ * JS module for Giphy API: https://www.npmjs.com/package/giphy-api
+ */
 
 exports.run = async (client, message, args, level) => {
   const start = Date.now();
-  const tags = args.join(" ");
-  giphy.random({
-    "tag": tags,
+  const query = args.join(" ");
+  giphy.search({
+    "q": query,
+    "limit": 100,
```

Comment:
Alternatively: Could we use offset and determine the bucket earlier? Not sure how large the results being returned is (or if it affects performance/memory)

Reply:
If we don't know how many total results there will be and we set the offset to a number too high, then no results will be returned. For example. 'ooga' returns 33 results. If we set the offset to 40, the results.data length is 0.
```
{
   "data":
   [
   ],
   "pagination":
   {
       "total_count": 8,
       "count": 0,
       "offset": 40
   },
   "meta":
   {
       "status": 200,
       "msg": "OK",
       "response_id": "z6iowljy8q5tvqz2s0mup20d4fkkbpc5tknb7alq"
   }
}
```

Reply:
Good catch. Nice catch on 35-36

Reply:
They call me the clutchmeister